libsolidity/formal/SMTEncoder.cpp

```diff
@@ -579,6 +579,28 @@ void SMTEncoder::endVisit(BinaryOperation const& _op)
 		);
 }
 
+bool SMTEncoder::visit(Conditional const& _op)
```

Comment:
I think this implementation works well for BMC, but it won't work for CHC if there's a function call in either expression.
In the end I think this needs to be similar to how each engine implements `IfStatement` separately. In that case, this body could be moved to BMC, and I could implement it for CHC in another PR.

Reply:
Hm, when I disabled BMC, was able to get correct results on conditional with function calls in both expressions...
Actually, when implementing this, I was looking at IfStatement implementation and did similar things, both here and in BMC

Reply:
Yes, I think that's fine for BMC, but if you look into CHC, you'll see that it implements its own visitor for `IfStatement` and doesn't call `SMTEncoder::visit(IfStatement)` like BMC does.

Reply:
Ah, I see. Understood completely, will update accordingly

Reply:
For some reason, if I move the body of this visitor to BMC, it fails one of the tests...
I see also `IfStatment` implementation has first part in BMC, than calls function from `SMTEncoder`. Was there any particular reason for this? (if you remember...)

Reply:
Sorry, I forgot to correct myself that your first implementation was already fine, basically doing what the visitor for `IfStatement` does